behind-the-tankard/Assets/Scripts/Environment/WorldController.cs

```
@@ -71,4 +81,37 @@ private void UpdateQuotas()
             quota.Amount *= quotaModifier;
         }
     }
+
+
+    public void OnEndOfDay()
+    {
+        if(curWorldEvent.duration <= _eventDaysPassed)
+        {
+            curWorldEvent = ChooseRandomWorldEvent();
+            _eventDaysPassed = 0;
+        }
+    }
+
+    private WorldEvent ChooseRandomWorldEvent()
+    {
+        float maxValue = 0;
+        
+        foreach(WorldEvent worldEvent in events)
+        {
+            maxValue += worldEvent.chance;
+        }
+
+        float rand = Random.Range(0, maxValue);
+        float previousValue = 0;
+        foreach(WorldEvent worldEvent in events)
+        {
+            if(rand > previousValue && rand < previousValue + worldEvent.chance)
+            {
+                return worldEvent;
+            }
+            previousValue += worldEvent.chance;
+        }
+        return null;
```

Throw an exception instead of returning null, because you don't expect to ever get here.